Examples/CaseStudies/SwiftUICaseStudies/01-GettingStarted-AlertsAndActionSheets.swift

```diff
@@ -0,0 +1,148 @@
+  Further, it automatically handles sending actions when you tap their buttons, which allows you \
+  to properly handle their functionality in the reducer rather than in two-way bindings and \
+  action closures.
+
+  The benefit of doing this is that you can get full test coverage on how a user interacts with \
+  with alerts and action sheets in your application
+  """
+
+struct AlertAndSheetState: Equatable {
+  var actionSheet = ActionSheetState<AlertAndSheetAction>.dismissed
+  var alert = AlertState<AlertAndSheetAction>.dismissed
+  var count = 0
+}
+
+enum AlertAndSheetAction: Equatable {
+  case actionSheetButtonTapped
+  case actionSheetCancelTapped
+  case alertButtonTapped
+  case alertCancelTapped
+  case decrementButtonTapped
+  case incrementButtonTapped
+}
+
+struct AlertAndSheetEnvironment {}
+
+let AlertAndSheetReducer = Reducer<
+  AlertAndSheetState, AlertAndSheetAction, AlertAndSheetEnvironment
+> { state, action, _ in
+
+  switch action {
+  case .actionSheetButtonTapped:
+    state.actionSheet = .show(
+      .init(
+        buttons: [
+          .init(
+            action: .actionSheetCancelTapped,
+            label: "Cancel",
+            type: .cancel
+          ),
+          .init(
+            action: .incrementButtonTapped,
+            label: "Increment"
+          ),
+          .init(
+            action: .decrementButtonTapped,
+            label: "Decrement"
+          ),
+        ],
+        message: "This is an action sheet.",
+        title: "Action sheet"
+      )
+    )
+    return .none
+
+  case .actionSheetCancelTapped:
+    state.actionSheet = .dismissed
+    return .none
+
+  case .alertButtonTapped:
+    state.alert = .show(
+      .init(
+        message: "This is an alert",
+        primaryButton: .init(
+          action: .alertCancelTapped,
+          label: "Cancel",
+          type: .cancel
+        ),
+        secondaryButton: .init(
+          action: .incrementButtonTapped,
+          label: "Increment"
+        ),
+        title: "Alert!"
+      )
+    )
+    return .none
+
+  case .alertCancelTapped:
+    state.alert = .dismissed
+    return .none
+
+  case .decrementButtonTapped:
+    state.actionSheet = .dismissed
+    state.count -= 1
+    return .none
+
+  case .incrementButtonTapped:
+    state.actionSheet = .dismissed
+    state.alert = .dismissed
+    state.count += 1
+    return .none
+  }
+}
+
+struct AlertAndSheetView: View {
+  let store: Store<AlertAndSheetState, AlertAndSheetAction>
+
+  var body: some View {
+    WithViewStore(self.store) { viewStore in
+      Form {
+        Section(header: Text(template: readMe, .caption)) {
+          Text("Count: \(viewStore.count)")
+
+          Button("Alert") { viewStore.send(.alertButtonTapped) }
+            .alert(
+              viewStore.alert,
+              send: viewStore.send,
+              dismissal: .alertCancelTapped
+            )
```

Comment:
this is the bit of API that I think can maybe be improved. Right now we need to pass along the state that represents the alert, the view store's `send` so that actions can be sent when interacting with buttons in the alert, and we further need a dismissal action to fire off when the alert is closed for non-button reasons (the system can close alerts without user interaction).